src/core/index.js

```diff
@@ -1,7 +1,75 @@
-module.exports = function(){
-  let cy = this;
+const popperRenderer = require('./render.js');
+const createReferenceObject = require('./createReferenceObject.js'); 
 
-  // your extension impl...
+//Create a popper object (This is for use on the core)
+module.exports.core = function (userOptions) {
+  //Get cytoscape object and container
+  var cy = this;
 
-  return this; // chainability
+  userOptions = resolveUndefined(userOptions); 
+
+  //Create popper object
+  var popper = popperRenderer.createPopperObject(cy, userOptions);
+
+  return popper;
+};
+
+//Create a popper object for first element in a collection
+module.exports.collection = function (userOptions) {
+  var elements = this;
+  var element = elements[0];
+  userOptions = resolveUndefined(userOptions); 
+
+  //Popper.js Should only be used on 1 element
+  if (elements.length > 1) {
+    console.log("Popper.js Extension should only be used on one element.");
```

Comment:
console.warn() may be a better fit

https://developer.mozilla.org/en-US/docs/Web/API/Console/warn